packages/connectivity/src/scp-cf/destination/service-binding-to-destination.ts

```diff
@@ -21,10 +27,48 @@ export const serviceToDestinationTransformers: Record<
   workflow: workflowBindingToDestination,
   'service-manager': serviceManagerBindingToDestination,
   xsuaa: xsuaaToDestination,
-  aicore: aicoreToDestiation
+  aicore: aicoreToDestination
 };
 
-async function aicoreToDestiation(
+/**
+ * Convenience function to create an OAuth2ClientCredentials destination from the provided service binding.
+ * If a JWT is provided as part of options, the tenant in the JWT is used for client credentials grant, else the provider tenant is used.
+ * Supported service types are:
+ * - business-logging
+ * - destination
+ * - s4-hana-cloud (falls back to basic authentication for this service type)
+ * - saas-registry
+ * - workflow
+ * - service-manager
+ * - xsuaa
+ * - aicore.
+ *
+ * Throws an error if the provided service binding is not supported.
+ * @param serviceBinding - The service binding to transform.
+ * @param options - Options used during fetching destinations;includes a JWT payload {@link JwtPayload} and options to influence caching behavior see {@link CachingOptions}.
+ * @returns A promise returning the transformed OAuth2ClientCredentials destination on success.
+ */
+export async function transformServiceBindingToDestination(
+  serviceBinding: Service,
+  options?: ServiceBindingTransformOptions
+): Promise<Destination> {
+  if (serviceToDestinationTransformers[serviceBinding.label]) {
```

Comment:
[q] For now I thought it is okay to return a Basic Auth destination only for 's4-hana-cloud' as it is documented well.

Should I throw an unsupported error for 's4-hana-cloud' instead?
WDY prefer?

Reply:
[q] Why do we even need a warning here? The binding wants basic, so it is basic. To me, this is expected behavior. Therefore I would suggest to remove the warning and also adjust the description as pointed out above.

Reply:
Added the auth flows in the description to make it more explicit.